### PR TITLE
Add CDF, Quantile, and Survival for Beta and StudentsT

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

**Comment by btracey** [comment]
Requires https://github.com/gonum/mathext/pull/21

File: distuv/studentst.go [hunk 1]
```diff
@@ -40,6 +42,24 @@ type StudentsT struct {
...
+	if y == 0 {
+		return 0.5
+	}
+	// For t > 0
+	// F(y) = 1 - 0.5 * I_t(y)(nu/2, 1/2)
+	// t(y) = nu/(y^2 + nu)
+	// and 1 - F(y) for t < 0
+	t := s.Nu / (y*y + s.Nu)
+	if y > 0 {
+		return 1 - 0.5*mathext.Incbeta(s.Nu/2, 0.5, t)
```
Comment (sbinet) [inline distuv/studentst.go]: why `s.Nu/2` and not `0.5*s.Nu` ?

File: distuv/beta.go [hunk 2]
```diff
@@ -88,6 +109,17 @@ func (b Beta) StdDev() float64 {
 	return math.Sqrt(b.Variance())
 }
 
+// Survival returns the survival function (complementary CDF) at x.
+func (b Beta) Survival(x float64) float64 {
+	if x <= 0 {
```
Comment (sbinet) [inline distuv/beta.go]: could perhaps be rewritten as a switch? (I personally find it more readable. but YMMV)

```go
switch {
case x <= 0:
    return 1
case x >= 1:
    return 0
}
```

File: distuv/beta.go [hunk 3]
```diff
@@ -27,6 +29,17 @@ type Beta struct {
...
 
+// CDF computes the value of the cumulative density function at x.
+func (b Beta) CDF(x float64) float64 {
+	if x <= 0 {
+		return 0
+	}
+	if x >= 1 {
+		return 1
+	}
+	return mathext.Incbeta(b.Alpha, b.Beta, x)
```
Comment (vladimir-ch) [inline distuv/beta.go]: Isn't this missing `/ mathext.Beta(b.Alpha, b.Beta)`?

Comment (btracey) [inline distuv/beta.go]: Hmm... why do the tests pass?

Comment (vladimir-ch) [inline distuv/beta.go]: My question is a real question. Looking at cephes I think that the factor is included in `Incbeta`. If that's the case, the documentation in `mathext` should be fixed.

Comment (btracey) [inline distuv/beta.go]: Yea, agreed. The documentation suggests it computes the unregularized integral, but these tests hopefully wouldn't pass if it did.

File: distuv/beta.go [hunk 4]
```diff
@@ -27,6 +29,17 @@ type Beta struct {
 	Source *rand.Rand
 }
 
+// CDF computes the value of the cumulative density function at x.
```
Comment (vladimir-ch) [inline distuv/beta.go]: s/density/distribution/

File: distuv/beta.go [hunk 5]
```diff
@@ -76,6 +89,14 @@ func (b Beta) Prob(x float64) float64 {
 	return math.Exp(b.LogProb(x))
 }
 
+// Quantile returns the inverse of the cumulative probability distribution.
```
Comment (vladimir-ch) [inline distuv/beta.go]: s/probability distribution/distribution function/

File: distuv/studentst.go [hunk 6]
```diff
@@ -40,6 +42,24 @@ type StudentsT struct {
 	Src *rand.Rand
 }
 
+// CDF computes the value of the cumulative density function at x.
```
Comment (vladimir-ch) [inline distuv/studentst.go]: s/density/distribution/

File: distuv/studentst.go [hunk 7]
```diff
@@ -69,6 +89,29 @@ func (s StudentsT) Prob(x float64) float64 {
 	return math.Exp(s.LogProb(x))
 }
 
+// Quantile returns the inverse of the cumulative probability distribution.
```
Comment (vladimir-ch) [inline distuv/studentst.go]: s/probability distribution/distribution function/

File: distuv/beta.go (hunk 3 above)
Comment (btracey) [inline distuv/beta.go]: The tests fail with the beta addition.

File: distuv/studentst.go (hunk 1 above)
Comment (btracey) [inline distuv/studentst.go]: Done.

File: distuv/beta.go (hunk 2 above)
Comment (btracey) [inline distuv/beta.go]: Done.

File: distuv/beta.go (hunk 5 above)
Comment (btracey) [inline distuv/beta.go]: Done.

File: distuv/studentst.go (hunk 6 above)
Comment (btracey) [inline distuv/studentst.go]: Done.

File: distuv/studentst.go (hunk 7 above)
Comment (btracey) [inline distuv/studentst.go]: Done.

**Pull request merged**